services/collaboration/pkg/connector/fileconnector.go

```diff
@@ -565,3 +567,19 @@ func (f *FileConnector) CheckFileInfo(ctx context.Context) (FileInfo, error) {
 	logger.Debug().Msg("CheckFileInfo: success")
 	return fileInfo, nil
 }
+
+func (f *FileConnector) ensureWatermarkText(user *userv1beta1.User) (watermarkText string) {
+	if f.cfg.Commons != nil {
+		watermarkText = f.cfg.Commons.OcisURL
+	}
+	if user != nil {
+		text := user.GetUsername() + " " + user.GetMail()
+		if strings.TrimSpace(text) != "" {
+			watermarkText = strings.TrimSpace(text)
+		}
+	}
+	if watermarkText == "" {
+		watermarkText = "Watermark"
+	}
```

Comment:
I think it's easier to either go for a "bottom to top" or "top to bottom" approach. This means that either you set a default value and start overwriting it (1), or you check and return the values you want (2)

Option 1:
```
watermark := "Watermark"
if Commons {
  watermark = Commons.OcisURL
}
if user {
  watermark = userText
}
return watermark
```

option 2:
```
if user {
  return userText
}
if Commons {
  return Commons.OcisURL
}
return "Watermark"
```

option 2 might be better because it should run less operations on average.